src/Resources/contao/dca/tl_content.php

```diff
@@ -716,9 +716,27 @@
 		(
 			'label'                   => &$GLOBALS['TL_LANG']['tl_content']['articleAlias'],
 			'exclude'                 => true,
-			'inputType'               => 'select',
-			'options_callback'        => array('tl_content', 'getArticleAlias'),
-			'eval'                    => array('mandatory'=>true, 'chosen'=>true, 'submitOnChange'=>true, 'tl_class'=>'w50 wizard'),
+			'inputType'               => 'picker',
+			//'options_callback'        => array('tl_content', 'getArticleAlias'),
+			'eval'                    => array(
+				'mandatory'    => true,
+				'tl_class'     => 'clr',
+
+				'foreignTable' => 'tl_article', // Foreign table for widget record fetching. @todo get from picker somehow?
```

Comment:
Imho fetching and validating the records should be done in simple, small methods so they can be easily overridden. Right now there's still to much going on in one single method. Can you split that up a little? Then that todo can be removed :)

Reply:
Do you mean some custom callbacks here or just the protected methods inside the widget class?

Reply:
Protected methods :)

Reply:
See 68d9b80.